clang/lib/CodeGen/CGDebugInfo.cpp

```diff
@@ -406,6 +406,17 @@ llvm::DIFile *CGDebugInfo::getOrCreateFile(SourceLocation Loc) {
 
   SmallString<32> Checksum;
 
+  if (CGM.getCodeGenOpts().SYCLUseMainFileName && FID.isInvalid() &&
+      !llvm::sys::path::is_absolute(FileName)) {
+    FileID MainFileID = SM.getMainFileID();
+    auto ExpectedFileRef = SM.getFileManager().getFileRef(FileName);
```

Comment:
It looks like you could potentially come through this code for some file other than the "main" file.

Reply:
DO you have an example in mind?

Reply:
No, I tried a few but didn't find one that would trigger this.

Reply:
I might potentially put this under a condition  like this: if(SM.getFileEntryForID(SM.getMainFileID())) I suppose, or add an assert to make sure that we catch it if it happens.